### PR TITLE
 Add AI TEDS create channel tests

### PR DESCRIPTION
**Pull request by zhindes**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This adds a test per TEDS AI create channel method to validate that they can successfully be used. There is a small amount of functional validation that parameters are properly being passed through the layers of Python and underlying driver API.

### Why should this Pull Request be merged?

This will give us confidence that we haven't broken anything as we refactor parts of the Python and driver stack.

### What testing has been done?

```
$ poetry run ni-python-styleguide fix tests && poetry run ni-python-styleguide lint tests && poetry run pytest -rs -k teds
=============================== test session starts ===============================
platform win32 -- Python 3.10.10, pytest-7.4.2, pluggy-1.3.0
rootdir: C:\dev\nidaqmx-python
configfile: pyproject.toml
testpaths: tests
plugins: cov-4.1.0, mock-3.11.1
collected 1431 items / 1357 deselected / 74 selected

tests\component\_task_modules\channels\test_ai_channel.py .................. [ 24%]
......................................                                       [ 75%]
tests\component\system\test_physical_channel.py ........                     [ 86%]
tests\component\system\test_physical_channel_properties.py ........          [ 97%]
tests\legacy\test_teds.py ..                                                 [100%]

======================= 74 passed, 1357 deselected in 3.03s =======================
```

### CODE REVIEW

File: tests/component/_task_modules/channels/test_ai_channel.py [hunk 1]
```diff
@@ -433,6 +618,20 @@ def test___task__add_ai_pos_lvdt_chan___sets_channel_attributes(
     assert chan.ai_ac_excit_freq == voltage_excit_freq
 
 
+def test___task__add_teds_ai_pos_lvdt_chan___sets_channel_attributes(
+    task: Task,
+    sim_position_device: Device,
+    lvdt_teds_file_path,
+):
+    # Our LVDT/RVDT TEDS files are invalid, but we can validate we got deep into the driver.
```
Comment (bkeryan) [inline tests/component/_task_modules/channels/test_ai_channel.py:626]: I think we should understand what is wrong with these files, or at least create a work item about doing so. The LV TEDS library is able to display these files.

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: tests/component/_task_modules/channels/test_ai_channel.py (hunk 1 above)
Comment (zhindes) [reply tests/component/_task_modules/channels/test_ai_channel.py:626]: I'll debug!, should be fun!

Comment (zhindes) [reply tests/component/_task_modules/channels/test_ai_channel.py:626]: Following up - there is a DAQ bug causing this behavior; the files are fine. I'll file that and see if there is another active device that supports these. If not, then this is the best I can do, but I'll fix the comment.

File: tests/conftest.py [hunk 2]
```diff
@@ -513,6 +513,110 @@ def interpreter(system: nidaqmx.system.System) -> BaseInterpreter:
 
 
 @pytest.fixture
-def teds_file_path(test_assets_directory):
+def teds_assets_directory(test_assets_directory):
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply tests/conftest.py]: Do the parameters too, not just the return value.

File: tests/conftest.py [hunk 3]
```diff
@@ -513,6 +513,110 @@ def interpreter(system: nidaqmx.system.System) -> BaseInterpreter:
...
+    """Returns the path to TEDS assets."""
+    return pathlib.Path(test_assets_directory, "teds")
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply tests/conftest.py]: These fixtures don't need to construct a pathlib.Path because the param is already one.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**